app/src/main/java/com/frantonlin/grocerylist/GroceryAdapter.java

```diff
@@ -0,0 +1,71 @@
+package com.frantonlin.grocerylist;
+
+import android.content.Context;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ArrayAdapter;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import java.util.ArrayList;
+
+/**
+ * Created by franton on 9/13/15.
+ */
+public class GroceryAdapter extends ArrayAdapter<GroceryItem> {
+    private ArrayList<GroceryItem> items;
+
+    /**
+     * Constructor
+     * @param context the context of the ArrayAdapter
+     * @param items the items handled by the ArrayAdapter
+     */
+    public GroceryAdapter(Context context, ArrayList<GroceryItem> items) {
+        super(context, 0, items);
+        this.items = items;
+    }
+
+    @Override
+    public View getView(final int position, View convertView, ViewGroup parent) {
+        // Get the information for this position
+        String itemName = getItem(position).getName();
+        Boolean checked = getItem(position).getChecked();
+
+        // Check if an existing view is being reused, otherwise inflate the view
+        if (convertView == null) {
+            convertView = LayoutInflater.from(getContext()).inflate(R.layout.grocery_item, parent, false);
+        }
+
+        // Look up view for information population
+        ImageView checkbox = (ImageView) convertView.findViewById(R.id.checkbox);
+        TextView list_item = (TextView) convertView.findViewById(R.id.list_item);
+        ImageView delete_icon = (ImageView) convertView.findViewById(R.id.delete_icon);
+
+        // Populate the information into the views
+        if(checked)
+            checkbox.setImageResource(R.drawable.ic_check_box_black_24dp);
+        else
+            checkbox.setImageResource(R.drawable.ic_check_box_outline_blank_black_24dp);
+        list_item.setText(itemName);
+
+        // Add onClickListeners
+        checkbox.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                items.get(position).toggleChecked();
+                notifyDataSetChanged();
+            }
+        });
+        delete_icon.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                items.remove(position);
+                notifyDataSetChanged();
+            }
+        });
+
+        // Return the completed view to render on screen
+        return convertView;
+    }
+}
```

Comment:
We haven't super talked about it yet, but generally you can just add an overarching method comment that explains the purpose of the method & get rid of the comments throughout. Also if you have to scroll to see the method declaration and the end of the method on your screen, the method is too long- once again, we haven't really talked about all that so no worries.